src/app/flights/flight-form/flight-form.component.html

```diff
@@ -1,79 +1,86 @@
-<form [formGroup]="form">
-  <div class="row">
-    <div class="col-sm-8">
-      <mat-form-field>
-        <input matInput formControlName="code" placeholder="Code" #codeInput>
-        <mat-error *ngIf="form.get('code')?.errors?.required">Code is required</mat-error>
-        <mat-error *ngIf="form.get('code')?.errors?.minlength">Minimum length is 4 chars</mat-error>
-        <mat-error *ngIf="form.get('code')?.errors?.maxlength">Maximum length is 7 chars</mat-error>
-        <mat-hint align="end"> {{ codeInput.value.length }} / 7</mat-hint>
-      </mat-form-field>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-sm-4">
-      <mat-form-field>
-        <input matInput formControlName="origin" placeholder="Origin">
-      </mat-form-field>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-sm-4">
-      <mat-form-field>
-        <input matInput formControlName="destination" placeholder="Destination">
-      </mat-form-field>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-sm-4">
-      <mat-form-field>
-        <input matInput formControlName="departureTime" placeholder="Departure time">
-      </mat-form-field>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-sm-4">
-      <mat-form-field>
-        <input matInput formControlName="returnTime" placeholder="Return time">
-      </mat-form-field>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-sm-8">
-      <textarea matInput formControlName="additionalInformation" placeholder="Additional Information"></textarea>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col">
-      <mat-checkbox formControlName="withSKPlanesDiscount">With WP Discount</mat-checkbox>
-    </div>
-  </div>
-  <div formArrayName="crew" class="crew-container shadow" *ngFor="let crewMember of crew.controls;index as index">
-    <div class="row crew-header">
-      <div class="col">
-        <span>Crew Member - {{ index + 1}} </span>
-        <mat-icon class="float-right" (click)="removeCrewMember(index)">delete_forever</mat-icon>
+<div class="app-flight-form">
+  <form [formGroup]="form">
+    <div class="app-flight-form__form-field">
+      <div>
+          <mat-form-field>
+            <input matInput formControlName="code" placeholder="Code" #codeInput>
+            <mat-error *ngIf="form.get('code')?.errors?.required">Code is required</mat-error>
```

Comment:
Zrób w pliku ts metodę `errors() { return this.form.get('code')?.errors; }` i tutaj w html napisz `errors?.required`.
Analogicznie w 2 przypadkach poniżej.